airflow/jobs/scheduler_job.py

```diff
@@ -880,13 +880,21 @@ def _run_scheduler_loop(self) -> None:
                     # is finished to avoid concurrent access to the DB.
                     self.log.debug("Waiting for processors to finish since we're using sqlite")
                     self.processor_agent.wait_until_finished()
-
-                with create_session() as session:
-                    num_queued_tis = self._do_scheduling(session)
-
-                    self.executor.heartbeat()
-                    session.expunge_all()
-                    num_finished_events = self._process_executor_events(session=session)
+                max_retry_count = conf.getint('scheduler', 'scheduler_loop_max_retries', fallback=5)
```

Comment:
What is this configuration? If you’re adding a new key, you need to add it to the configuration declaration. See files in `airflow/config_templates`.

Reply:
Since I use fallback it's not mandatory, do you prefer I add it to config template for documentation

Reply:
Yes. Optional keys should be added as a commented out value (there are several examples) so they can be discovered by users.

Reply:
Why new config?

We already have `airflow.utils.retries.run_with_db_retries`

Reply:
that get's the number of retries from `conf.getint('database', 'max_db_retries', fallback=3)`

Reply:
Since upgrading to 2.x (we've tried multiple versions up to 2.3.1) we've been having many sporadic "MySql has gone away" exceptions causing the scheduler to either hang or crash. and since adding this simple retry our scheduler had zero such limbo cases for weeks

Reply:
Do you prefer I reuse the same conf.getint('database', 'max_db_retries', fallback=3)?